homeassistant/components/madvr/coordinator.py

```diff
@@ -0,0 +1,80 @@
+"""Coordinator for handling data fetching and updates."""
+
+import logging
+
+from madvr.madvr import Madvr
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class MadVRCoordinator(DataUpdateCoordinator):
```

Comment:
Please type the coordinator to the type of `data`
```suggestion
class MadVRCoordinator(DataUpdateCoordinator[..]):
```